git_repo/services/service.py

```diff
@@ -272,24 +272,25 @@ def _convert_user_into_remote(self, username, exclude=['all']):
                     if self.fqdn in url and username == url.split('/')[-2].split(':')[-1]:
                         yield name
 
-    def format_path(self, repository, namespace=None, rw=False):
+    def format_path(self, repository, namespace=None, rw=False, allow_no_namespace=False):
```

Comment:
I've had a thought about this, and the same way we can allow for a [maximum of nested namespaces](https://github.com/guyzmo/git-repo/pull/183/commits/623f79a1b5ab7a3eb2df48f4f50c46803524758b#diff-c2d533c7e9ec06708c3d3ef358a65e4bR13), we could setup a minimum as a class member which has the default of 2 for all services? And then here check that `repo.count('/') >= self._min_nested_namespaces-1`? I feel like that'd be a more generic way of handling this. But maybe I'm complicating things.

I guess the pros of the above option would be to keep the prototype simpler and factorize behaviour in a way homogeneous with `_max_nested_namespaces`. The cons would be to use a side effect to modify the method's behaviour.

@Phantom-42, @jayvdb : your opinions?

Reply:
I like this idea. Anyway, it would be interesting for me to hear @jayvdb opinion before changing this.

Reply:
And one more thing. I think we better set default `_min_nested_namespaces` to 1, and then check with `repo.count('/') >= self._min_nested_namespaces`. I think that makes a bit more sense and avoids having such confusing code as:
```
_max_nested_namespaces = 1
_min_nested_namespaces = 2
```

@guyzmo What do you think about that?

Reply:
> @guyzmo What do you think about that?

Indeed, I do agree with you, that makes sense. I the above context I forgot that namespaces were excluding the project name ☺

Reply:
I totally agree with you on that! That makes sense!

In the above context I forgot that namespaces were excluding the project name 😉

Reply:
Okay. I changed the code to use `_min_nested_namespaces` now